libsolidity/interface/CompilerStack.cpp

```diff
@@ -1246,28 +1246,28 @@ void CompilerStack::storeContractDefinitions()
 
 void CompilerStack::annotateInternalFunctionIDs()
 {
-	uint64_t internalFunctionID = 1;
 	for (Source const* source: m_sourceOrder)
 	{
 		if (!source->ast)
 			continue;
 
 		for (ContractDefinition const* contract: ASTNode::filteredNodes<ContractDefinition>(source->ast->nodes()))
 		{
+			uint64_t internalFunctionID = 1;
 			ContractDefinitionAnnotation& annotation = contract->annotation();
 
 			if (auto const* deployTimeInternalDispatch = util::valueOrNullptr((*annotation.deployedCallGraph)->edges, CallGraph::SpecialNode::InternalDispatch))
 				for (auto const& node: *deployTimeInternalDispatch)
 					if (auto const* callable = get_if<CallableDeclaration const*>(&node))
 						if (auto const* function = dynamic_cast<FunctionDefinition const*>(*callable))
-							if (!function->annotation().internalFunctionID.set())
-								function->annotation().internalFunctionID = internalFunctionID++;
+							if (!contract->annotation().internalFunctionIDs[function].set())
```

Comment:
Is this actually ever false in this version? I.e. can it ever happen that we already have set an internal function ID for a function in the contract here?
If not, we can turn this into an assertion (``contract->annotation().internalFunctionIDs.count(function) == 0``). Otherwise, we can just check ``contract->annotation().internalFunctionIDs.count(function) == 0`` instead after removing the SetOnce.